src/main/java/codecain/GraphicalUserInterface/View/ClassNode.java

```diff
@@ -513,6 +521,45 @@ public boolean removeMethod(String methodName) {
         return removed;
     }
 
+    /**
+     * Updates the preferred width of the {@code ClassNode} to ensure it can accommodate
+     * the longest text in the {@code fields} and {@code methods} {@link ListView}.
+     * <p>
+     * The method calculates the maximum text width of items in both {@code fields}
+     * and {@code methods} by measuring their rendered width using the {@link Text} class.
+     * It then adjusts the preferred width of the {@code ClassNode} to be the greater
+     * of the calculated maximum width plus padding or a predefined minimum width.
+     * </p>
+     * <p>
+     * This ensures the {@code ClassNode} resizes dynamically as items are added to
+     * or removed from the {@code fields} and {@code methods} lists.
+     * </p>
+     */
+    private void updateWidth() {
+        // Calculate the widest text in fields
+        double maxFieldWidth = fields.getItems().stream()
+                .map(field -> new Text(field.toString()).getLayoutBounds().getWidth())
+                .max(Double::compare)
+                .orElse(0.0); // Default to 0 if no fields are present
+
+        // Calculate the widest text in methods
+        double maxMethodWidth = methods.getItems().stream()
+                .map(method -> new Text(method.toString()).getLayoutBounds().getWidth())
+                .max(Double::compare)
+                .orElse(0.0); // Default to 0 if no methods are present
+
+        // Determine the maximum width needed based on the longer text
+        double maxContentWidth = Math.max(maxFieldWidth, maxMethodWidth) + 50; // Add padding
+
+        // Ensure the width is at least MIN_WIDTH
+        double finalWidth = Math.max(maxContentWidth, MIN_WIDTH);
+
+        // Set the new preferred width for the ClassNode
+        this.setPrefWidth(finalWidth);
+
+        System.out.println("Updated width to: " + finalWidth);
+    }
+
```

Comment:
This looks good to me. I like how you set min values for width and height instead of having set values. The dynamic adjusting looks good to me.